modular_skyrat/code/game/machinery/cryptominer/cryptominers.dm

```diff
@@ -44,59 +42,68 @@
 		return
 
 /obj/machinery/cryptominer/process()
-	var/turf/L = loc
-	var/datum/gas_mixture/env = L.return_air()
-	if(env.temperature > maxtemp)
+	var/turf/T = get_turf(src)
+	if(!T)
+		return
+	var/datum/gas_mixture/env = T.return_air()
+	if(!env)
+		return
+	if(!mining)
+		return
+	if(env.temperature >= maxtemp)
 		if(mining)
-			playsound(loc, 'sound/machines/beep.ogg', 50, 1, -1)
-		mining = FALSE
-		update_icon()
+			playsound(loc, 'sound/machines/beep.ogg', 50, TRUE, -1)
+		set_mining(FALSE)
 		return
-	if(env.temperature < maxtemp && env.temperature > midtemp)
+	if(env.temperature <= maxtemp && env.temperature >= midtemp)
 		if(mining)
-			playsound(loc, 'sound/machines/ping.ogg', 50, 1, -1)
-			var/datum/bank_account/D = SSeconomy.get_dep_account(ACCOUNT_CAR)
-			if(D)
-				D.adjust_money((miningpoints / 5))
-			env.temperature += 100
-			air_update_turf()
-	if(env.temperature < midtemp && env.temperature > mintemp)
+			produce_points(0.20)
+			produce_heat()
+		return
+	if(env.temperature <= midtemp && env.temperature >= mintemp)
 		if(mining)
-			playsound(loc, 'sound/machines/ping.ogg', 50, 1, -1)
-			var/datum/bank_account/D = SSeconomy.get_dep_account(ACCOUNT_CAR)
-			if(D)
-				D.adjust_money((miningpoints))
-			env.temperature += 100
-			air_update_turf()
+			produce_points(1)
+			produce_heat()
+		return
 	if(env.temperature <= mintemp)
 		if(mining)
-			playsound(loc, 'sound/machines/ping.ogg', 50, 1, -1)
-			var/datum/bank_account/D = SSeconomy.get_dep_account(ACCOUNT_CAR)
-			if(D)
-				D.adjust_money((miningpoints * 3))
-			env.temperature += 100
-			air_update_turf()
+			produce_points(3)
+			produce_heat()
+		return
+
+/obj/machinery/cryptominer/proc/produce_points(number)
+	playsound(loc, 'sound/machines/ping.ogg', 50, TRUE, -1)
+	var/datum/bank_account/D = SSeconomy.get_dep_account(ACCOUNT_CAR)
+	if(D)
+		D.adjust_money(FLOOR(miningpoints * number,1))
+
+/obj/machinery/cryptominer/proc/produce_heat()
+	atmos_spawn_air("co2=10;TEMP=2000")
```

Comment:
Makes co2 out of nowhere? This may lead to ridiculous pressures even though the room is being kept cooled, as the gas keeps being created.

Reply:
Actually this doesnt need to make sense, I guess it's better for gameplay that way